1) Load Data:
    Feature Engineering with Weather Data

In [1]:
import pandas as pd 

In [5]:
# Ruta al archivo CSV
file_path = file_path = r'C:\Users\asanchezbelb\Desktop\WAREHOUSE\data_jena_climate_2009_2016\jena_climate_2009_2016.csv'

# Cargar el archivo CSV en un DataFrame
df = pd.read_csv(file_path, header=0, index_col=0)

df.head()

p (mbar)  T (degC)  Tpot (K)  Tdew (degC)  rh (%)  \
Date Time                                                                
01.01.2009 00:10:00    996.52     -8.02    265.40        -8.90    93.3   
01.01.2009 00:20:00    996.57     -8.41    265.01        -9.28    93.4   
01.01.2009 00:30:00    996.53     -8.51    264.91        -9.31    93.9   
01.01.2009 00:40:00    996.51     -8.31    265.12        -9.07    94.2   
01.01.2009 00:50:00    996.51     -8.27    265.15        -9.04    94.1   

                     VPmax (mbar)  VPact (mbar)  VPdef (mbar)  sh (g/kg)  \
Date Time                                                                  
01.01.2009 00:10:00          3.33          3.11          0.22       1.94   
01.01.2009 00:20:00          3.23          3.02          0.21       1.89   
01.01.2009 00:30:00          3.21          3.01          0.20       1.88   
01.01.2009 00:40:00          3.26          3.07          0.19       1.92   
01.01.2009 00:50:00          3.27          3.08          0.19       1.92   

                     H2OC (mmol/mol)  rho (g/m**3)  wv (m/s)  max. wv (m/s)  \
Date Time                                                                     
01.01.2009 00:10:00             3.12       1307.75      1.03           1.75   
01.01.2009 00:20:00             3.03       1309.80      0.72           1.50   
01.01.2009 00:30:00             3.02       1310.24      0.19           0.63   
01.01.2009 00:40:00             3.08       1309.19      0.34           0.50   
01.01.2009 00:50:00             3.09       1309.00      0.32           0.63   

                     wd (deg)  
Date Time                      
01.01.2009 00:10:00     152.3  
01.01.2009 00:20:00     136.1  
01.01.2009 00:30:00     171.6  
01.01.2009 00:40:00     198.0  
01.01.2009 00:50:00     214.3

Fecha y hora: Referencia de fecha y hora en el formato dd.mm.aaaa hh:mm:ss.

p (mbar): 996.52 mbar (milibar). El milibar es la unidad derivada del SI de presión utilizada para cuantificar la presión interna. Los informes meteorológicos suelen indicar la presión atmosférica en milibares.

T (degC): -8.02 grados Celsius. La temperatura en grados Celsius.

Tpot (K): 265.4 Kelvin. La temperatura en grados Kelvin.

Tdew (degC): -8.9 grados Celsius. La temperatura en grados Celsius en relación con la humedad. El punto de rocío es una medida de la cantidad absoluta de agua en el aire; el punto de rocío es la temperatura a la cual el aire no puede retener toda la humedad y el agua se condensa.

rh (%): 93.3%. La humedad relativa es una medida de cuán saturado está el aire con vapor de agua; el %HR determina la cantidad de agua contenida en objetos de colección.

VPmax (mbar): 3.33 mbar. La presión de vapor de saturación.

VPact (mbar): 3.11 mbar. La presión de vapor.

VPdef (mbar): 0.22 mbar. La deficiencia de presión de vapor.

sh (g/kg): 1.94 g/kg. La humedad específica.

H2OC (mmol/mol): 3.12 mmol/mol. La concentración de vapor de agua.

rho (g/m^3): 1307.75 g/m^3. Densidad del aire.

wv (m/s): 1.03 m/s. Velocidad del viento.

max. wv (m/s): 1.75 m/s. Velocidad máxima del viento.

wd (deg): 152.3 grados. Dirección del viento en grados.

In [8]:
def report(df):
    print("Nulos en el DataFrame:")
    print(df.isnull().sum())
    
    print("\nForma del DataFrame:")
    print(df.shape)
    
    print("\nTipos de datos en el DataFrame:")
    print(df.dtypes)
    
    print("\nDescripción del DataFrame:")
    print(df.describe().T)


report(df)

Nulos en el DataFrame:
p (mbar)           0
T (degC)           0
Tpot (K)           0
Tdew (degC)        0
rh (%)             0
VPmax (mbar)       0
VPact (mbar)       0
VPdef (mbar)       0
sh (g/kg)          0
H2OC (mmol/mol)    0
rho (g/m**3)       0
wv (m/s)           0
max. wv (m/s)      0
wd (deg)           0
dtype: int64

Forma del DataFrame:
(420551, 14)

Tipos de datos en el DataFrame:
p (mbar)           float64
T (degC)           float64
Tpot (K)           float64
Tdew (degC)        float64
rh (%)             float64
VPmax (mbar)       float64
VPact (mbar)       float64
VPdef (mbar)       float64
sh (g/kg)          float64
H2OC (mmol/mol)    float64
rho (g/m**3)       float64
wv (m/s)           float64
max. wv (m/s)      float64
wd (deg)           float64
dtype: object

Descripción del DataFrame:
                    count         mean        std      min      25%      50%  \
p (mbar)         420551.0   989.212776   8.358481   913.60   984.20   989.58   
T (degC)         42055